libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificatePinnerFactory.java

```diff
@@ -36,11 +37,37 @@ CertificatePinner provideCertificatePinnerFor(Environment environment) {
     return CERTIFICATES_PINS.get(environment);
   }
 
-  private void addCertificatesPins(Map<String, List<String>> pins, CertificatePinner.Builder builder) {
+  private void addCertificatesPins(Map<String, List<String>> pins, CertificatePinner.Builder builder,
+                                   CertificateBlacklist certificateBlacklist) {
+    pins = removeBlacklistedPins(pins, certificateBlacklist);
+
     for (Map.Entry<String, List<String>> entry : pins.entrySet()) {
       for (String pin : entry.getValue()) {
         builder.add(entry.getKey(), pin);
       }
     }
   }
+
+  private Map<String, List<String>> removeBlacklistedPins(Map<String, List<String>> pins,
+                                                          CertificateBlacklist certificateBlacklist) {
+    Set<String> pinsKey = pins.keySet();
+    String key = pinsKey.iterator().next();
+
+    List<String> hashList = pins.get(key);
+
+    List blackList = certificateBlacklist.retrieveBlackList();
```

Comment:
Shouldn't the block of code between line 58 and 62 be called first thing within this method?